### PR TITLE
Fix SocketsHttpHandler metrics test on Android

### PR DESCRIPTION
**Pull request by MihaZupan**
Saw failures like these in a few `runtime-extra-platforms` runs.
My guess is the test ended up using the mobile platform handler instead of `SocketsHttpHandler`.

```
Assert.Collection() Failure
Collection: [http.client.active_requests=1 [url.scheme=http, server.address=127.0.0.1, server.port=42873, http.request.method=GET], http.client.active_requests=-1 [url.scheme=http, server.address=127.0.0.1, server.port=42873, http.request.method=GET], http.client.request.duration=0.0143246 [url.scheme=http, server.address=127.0.0.1, server.port=42873, http.request.method=GET, http.response.status_code=200, network.protocol.version=1.1]]
Expected item count: 11
Actual item count:   3
   at System.Net.Http.Functional.Tests.HttpMetricsTest.<>c__DisplayClass19_0.<<AllSocketsHttpHandlerCounters_Success_Recorded>b__0>d.MoveNext() in /_/src/libraries/System.Net.Http/tests/FunctionalTests/MetricsTest.cs:line 618
--- End of stack trace from previous location ---
   at System.Threading.Tasks.TaskTimeoutExtensions.GetRealException(Task task) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 120
--- End of stack trace from previous location ---
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 90
   at System.Net.Test.Common.LoopbackServerFactory.<>c__DisplayClass6_0.<<CreateClientAndServerAsync>b__0>d.MoveNext() in /_/src/libraries/Common/tests/System/Net/Http/GenericLoopbackServer.cs:line 44
--- End of stack trace from previous location ---
   at System.Net.Test.Common.LoopbackServer.CreateServerAsync(Func`2 funcAsync, Options options) in /_/src/libraries/Common/tests/System/Net/Http/LoopbackServer.cs:line 107
   at System.Net.Http.Functional.Tests.HttpMetricsTest.AllSocketsHttpHandlerCounters_Success_Recorded() in /_/src/libraries/System.Net.Http/tests/FunctionalTests/MetricsTest.cs:line 575
```

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/FunctionalTests/MetricsTest.cs [hunk 1]
```diff
@@ -974,6 +974,12 @@ public void AllSocketsHttpHandlerCounters_Success_Recorded()
                 using HttpMetricsTest_DefaultMeter test = new(null);
                 await test.LoopbackServerFactory.CreateClientAndServerAsync(async uri =>
                 {
+                    if (PlatformDetection.IsMobile)
+                    {
+                        // Force the test to use SocketsHttpHandler
+                        AppContext.SetSwitch("System.Net.Http.UseNativeHttpHandler", false);
+                    }
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/tests/FunctionalTests/MetricsTest.cs:981]: Nit: should we move this to be the very first thing inside the Invoke callback?

**Review by stephentoub — APPROVED** [review]

**Comment by MihaZupan** [comment]
Thanks @simonrozsival, you're right, it was using the correct handler.
The actual issue was that we weren't flowing down the meter factory correctly - I opened #90298.

**Pull request closed without merging**